### PR TITLE
Ocean indices branch: OSI, PI and FAI

### LINKED ISSUE #50: Question about application domain and equation parameters
Hi,

I want to contribute with spectral indexes that are used to detect features at the surface of the ocean: oil spills (OSI), floating vegetation (FAI), floating debris (FDI, PI), ... Should I include these in the water domain? Should Floating Algae Index (FAI) be included in the vegetation domain, although it is applied on the ocean?

Also, FAI and FDI (Floating Debris Index) depend on the NIR and Red Edge 2 wavelengths that change between Sentinel-2A and Sentinel-2B. I see that lambdaN is already defined, but not lambdaRE2, does these parameters take into consideration Sentinel-2 A and B different values?

### PR DESCRIPTION
**Pull request by EmanuelCastanho**
Added to the water domain: Oil Spill Index, Plastic Index, and Floating Algae Index.

PI spectral index is identical to IPVI, but the applications are different. I do not know how you handle these cases.

I added lambdaS1 because of FAI. This spectral index also uses lambdaR so I think it makes more sense to change the description of the lambdas. I changed the description to central wavelength in different files.

Tests OK.

Closes #50

### CODE REVIEW

**Review by davemlz — CHANGES_REQUESTED** [review]
Nice work!

Please just update the DOI of OSI and it will be ready to be approved! ;)

**Comment by EmanuelCastanho** [comment]
You are right, it is giving `Error-Cannot Connect to Server`. The DOI comes from [ResearchGate](https://www.researchgate.net/publication/355541108_Oil_Spill_Index_OSI_to_Sentinel-2_Satellite_Data_QU_in_International_Contribution).
There is this one [https://doi.org/10.1016/j.mex.2021.101327](https://doi.org/10.1016/j.mex.2021.101327), it is a paper from the same author, uses the same band ratios (see Conclusion), but does not explicitly mention OSI.

I suggest:
a) Using this different paper;
b) Commenting this index until the OSI DOI is back again.

Thanks :)

**Comment by davemlz** [comment]
I just checked. The first DOI is for the poster. In the poster the authors reference the new DOI you sent. I also checked the Conclusions and there it is the ratio. I think it's fine if we keep that last DOI and make the index available ;)

Also, there are conflicts in `indices.py`, this is because new indices were just added, but not a major problem, just sync the branch ;)

**Comment by EmanuelCastanho** [comment]
Please confirm if everything is ready :)

File: src/indices.py [hunk 1]
```diff
@@ -2177,6 +2177,32 @@ class SpectralIndices(BaseModel):
             application_domain="vegetation",
             date_of_addition="2024-05-16",
             contributor="https://github.com/davemlz",
```
Comment (davemlz) [inline src/indices.py:2179]: Closing parenthesis is missing ;)

```suggestion
            contributor="https://github.com/davemlz",
        )
```

Comment (davemlz) [reply src/indices.py:2179]: Ups, forgot a comma:

```suggestion
            contributor="https://github.com/davemlz",
        ),
```

**Review by davemlz — CHANGES_REQUESTED** [review]
Just a small error. I added a suggestion, you can commit it and it will be done! ;)

**Comment by davemlz** [comment]
I realized I could commit it myself. I just did.

**Review by davemlz — APPROVED** [review]

**Comment by davemlz** [comment]
Merged!

The indices will be available in the new version of ASI. Thank you very much for your contribution!

Cheers,

David

**Pull request merged**